client/trino-cli/src/test/java/io/trino/cli/TestClientRedirect.java

```diff
@@ -0,0 +1,191 @@
+import io.trino.client.ClientSession;
+import io.trino.client.ClientTypeSignature;
+import io.trino.client.Column;
+import io.trino.client.QueryResults;
+import io.trino.client.StatementStats;
+import io.trino.client.uri.PropertyName;
+import io.trino.client.uri.TrinoUri;
+import okhttp3.logging.HttpLoggingInterceptor;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.sql.SQLException;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.Properties;
+
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.HttpHeaders.LOCATION;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.cli.ClientOptions.OutputFormat.CSV;
+import static io.trino.cli.TerminalUtils.getTerminal;
+import static io.trino.cli.TestQueryRunner.createClientSession;
+import static io.trino.client.ClientStandardTypes.BIGINT;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+@TestInstance(PER_METHOD)
+public class TestClientRedirect
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+    private static final String DUMMY_ACCESS_TOKEN = Base64.getEncoder().encodeToString("jwt-signing-key".getBytes(US_ASCII));
+    private MockWebServer server;
+
+    @BeforeEach
+    public void setup()
+            throws IOException
+    {
+        server = new MockWebServer();
+        server.start();
+    }
+
+    @AfterEach
+    public void teardown()
+            throws IOException
+    {
+        server.close();
+        server = null;
+    }
+
+    @Test
+    public void testDisableClientRedirectsDefault()
+            throws Exception
+    {
+        server.enqueue(new MockResponse()
+                .setResponseCode(307)
+                .addHeader(LOCATION, server.url("redirect/v1/statement"))
+                .addHeader(AUTHORIZATION, DUMMY_ACCESS_TOKEN));
+        server.enqueue(new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(createResults(server)));
+
+        QueryRunner queryRunner = createQueryRunner(createTrinoUri(server, Optional.empty()), createClientSession(server));
+
+        try (Query query = queryRunner.startQuery("SELECT 1")) {
+            query.renderOutput(getTerminal(), nullPrintStream(), nullPrintStream(), CSV, Optional.of(""), false);
```

Comment:
Can we assert the query result instead of trying to render the result? Same for others.

Reply:
Could you please share an example unit test where this is happening, couldn't find one.